### PR TITLE
feat/deena-emma/journal popup

### LINKED ISSUE #6: [US #2] Journal task Pop up
## Description

See MIRO board for Journal Task Setup. Have the Journal have a title, and a textbox. Offer the ability to **blur** the background when the Journal is active(have this be a property of the webpage like a "is" open. Use standard elements as much as possible. 
## Frontend

Have a dummy Webpage that contains boxes that are Journal Objects. When clicked on, this should open a popup with the **Title** **Date** **Description** and a bunch of smiley faces/emojis at the bottom(See Miro). The X button should not store the object. 

Only when submit is clicked(highlight this in green) have the object be rendered on the web page with its **Title** and a brief 40 character description in gray displayed underneath(see Miro). 
## Backend

Store Journal Objects in Local Storage and Retrieve them when they are clicked from the webpage.

### PR DESCRIPTION
**Pull request by faquanly**
close #6 
# Documentation For Journal Popup (updated 5/22)

## Role of Popup

The role of this component is to have a journal popup feature so that the user can input a journal entry in their dev notebook. 

image

## Intended Purpose

The journal popup is intended to open whenever the user clicks add new journal. A popup will display and the user has the ability to name their journal, give it a date, and add in their actual journal entry. Once they finish they can click the "done" button to save, or the "x" button to cancel. 

Note: 
- on submit button there is an entry_id, entry_title, entry_content, and creation_date saved to localStorage (and outputted on console)
- (EDITED) to follow same formatting and styling as the task popup

We implemented this through a seperate HTML, JS, and CSS file. 

## Requirements

The popup should allow the user to input information for each journal entry including date, title, and content. The background should blur to improve readability and clarity. It should be simple and straightforward to use from a UI perspective.

## Testing

For testing our journal popup component, we used the LiveServer to periodically check any changes that were made and the effects of those changes.

#### Authors

- Deena Pederson

- Emma Nguyen

### CODE REVIEW

File: source/project/components/app.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  }
+
+  /**
+   * @method connectedCallback
+   * @description Lifecycle method that is called when the component is inserted into the DOM.
+   * It sets up event listeners for form submission within the shadow DOM.
+   */
+  connectedCallback() {
+    // event listener to form submission
+    setTimeout(() => {
```
Comment (deenap23) [inline source/project/components/app.js]: this code was being called before the html was put into the DOM by the constructor so I added a timeout. Is there a better way to do this?

File: source/project/components/app.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      this.shadowRoot
+        .querySelector("#journalForm")
+        .addEventListener("submit", this.onSubmit.bind(this));
+    }, 3000);
+
+    // const closeButton = document.getElementById("closeBtn"); old
```
Comment (deenap23) [inline source/project/components/app.js]: In progress, working on getting the close button functionality (same issue that hashim and Matteo had)

**Review by mialyssa — APPROVED** [review]
Functionality wise, from Monday's meeting everything looks great and works as expected! The last documentation update is from 5/18, so I am not sure if it reflects the most recent stylistic changes; namely the done button alignment. Otherwise, It looks good to merge!

File: source/project/components/app.js (hunk 1 above)
Comment (SamvritSrinath) [reply source/project/components/app.js]: Maybe have another function that puts the html into the DOM, not sure whether this was resolved?

**Review by n2singh — APPROVED** [review]
Everything looks good, the styling is up to par with the other formats, unless there is something I am missing I think we are good to merge!

**Comment by SamvritSrinath** [comment]
Just an FYI —> vscode files should be ignored and not pushed to the main branch(I see that you are using it for testing, the GitHub tests will not run on the cloud). 

functionally things seem to be fine but we shouldn’t be fetching components from the frontend(I see that you are using the fetch API rather than creating a shadow component, could you embed this in the app.js and rename that to journal-popup.js and embed both this and the stylesheet within a folder?

**Review by SamvritSrinath — CHANGES_REQUESTED** [review]
Refactor Code structure according to latest comment —> components should be placed in their own folders with their own stylesheets. VScode files should be removed.

**Comment by faquanly** [comment]
We changed `app.js` to `journal-popup.js` and added the unit test `journal-popup.test.js` in `./__test__ /` directory
Screenshot 2024-05-23 at 7 12 57 PM

**Review by arnavkamdar — COMMENTED** [review]
I think it looks pretty good apart from the vs code file like Samvrit mentioned. I'd also add that the test HTML you use should probably be named a little more formally than testingDeena.html, but that's a pretty minor issue to fix.

**Review by SamvritSrinath — APPROVED** [review]

**Review by SamvritSrinath — APPROVED** [review]

**Pull request merged**